auth/server/base/src/main/java/org/wildfly/security/authz/RegexRoles.java

```diff
@@ -0,0 +1,82 @@
+package org.wildfly.security.authz;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import java.util.stream.StreamSupport;
+
+import static org.wildfly.security.auth.server._private.ElytronMessages.log;
+
+/**
+ * A regex roles.
+ * <p>
+ * Roles that are checked against regex and if matching are mapped with given replacement pattern. Regex can capture groups that replacement pattern can make use of.
+ *
+ * @author <a href="mailto:dvilkola@redhat.com">Diana Vilkolakova</a>
+ */
+public class RegexRoles implements Roles {
+    private final Roles delegate;
+    private final Pattern regex;
+    private final String replace;
+    private final Boolean keepNonMapped;
+
+    RegexRoles(final Roles delegate, Pattern regex, final String replace, final Boolean keepNonMapped) {
+        this.delegate = delegate;
+        this.regex = regex;
+        this.replace = replace;
+        this.keepNonMapped = keepNonMapped;
+    }
+
+    public boolean contains(final String roleName) {
+        try {
+            return StreamSupport.stream(delegate.spliterator(), false).anyMatch(role ->
+                    (regex.matcher(role).matches() && role.replaceFirst(regex.pattern(), replace).equals(roleName)) ||
```

Comment:
I think for some components where we allow a replacement to be specified, we also allow a boolean value like ```replaceAll``` to be configured that indicates whether all occurrences should be replaced or just the first. It can default to ```false``` (i.e., only replace the first occurrence by default). (e.g., see ```RegexNameRewriter```)

Reply:
I added `replaceAll` option. Thank you